client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -52,9 +50,61 @@ export class SiteVerticalsSuggestionSearch extends Component {
 		this.state = {
 			railcar: this.getNewRailcar(),
 			candidateVerticals: [],
+			isSuggestionSelected: false,
+			searchValue: props.searchValue,
 		};
+		this.updateVerticalDataDebounced = props.bindDebounce( this.updateVerticalData );
 	}
 
+	componentDidUpdate( prevProps ) {
+		// The suggestion list should only be updated when a user is not navigating the list through keying.
+		// Note: it's intentional to use object reference comparison here.
+		// Since `verticals` props is connected from a redux state here, if the two references are identical,
+		// we can safely say that the two content are identical, thanks to the immutability invariant of redux.
+		if ( prevProps.verticals !== this.props.verticals && ! this.state.isSuggestionSelected ) {
+			// It's safe here to call setState() because we prevent the indefinite loop by the wrapping condition.
+			// See the official doc here: https://reactjs.org/docs/react-component.html#componentdidupdate
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setSearchResults( this.props.verticals );
+		}
+	}
+
+	/***
+	 * Sets `state.results` with incoming HTTP results, retaining previous non-user vertical search results
+	 * if the incoming HTTP results contain only user-defined results.
+	 *
+	 * Then we'll debounce the update on the vertical while the user is typing to reduce
+	 * the amount of times we update the site preview.
+	 *
+	 * This function could better be performed in the backend eventually.
+	 *
+	 * @param {Array} results Incoming HTTP results
```

Comment:
One major purpose of the existence of the data layer, redux thunks or any other effectful data request paradigm is to isolate app state and UI components from the data source. In this case, `setSearchResults()` shouldn't be aware of the HTTP thing, but only the data shape of `results`. Thus, let's get rid of all the HTTP-related things in the description :)